src/transformState.js

```diff
@@ -4,8 +4,39 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+const removeKey = (action, data) => {
+  const { keysToRemove } = action;
+
+  for (const keyToRemove of keysToRemove) {
+    delete data[keyToRemove];
+  }
+};
+const clearProperties = (action, data) => {
+  for (const key in data) {
+    delete data[key];
+  }
+};
+
 function transformState(state, actions) {
-  // write code here
+  const stateCopy = state;
+
+  for (const action of actions) {
+    const { type, extraData } = action;
+
+    switch (type) {
+      case 'clear':
+        clearProperties(action, stateCopy);
+        break;
+      case 'addProperties':
+        Object.assign(stateCopy, extraData);
+        break;
+      case 'removeProperties':
+        removeKey(action, stateCopy);
+        break;
+    }
```

Comment:
You forgot to make a `default` state with an error when `action.type` is unknown.
Try to use `throw new Error()` and describe error with `action.type`.
`[CODE STYLE]: switch/case should always have default case for error handling.`

Reply:
yes. added now